revocation/revocation.go

```diff
@@ -44,10 +64,12 @@ func New(httpClient *http.Client) (Revocation, error) {
 	if httpClient == nil {
 		return nil, errors.New("invalid input: a non-nil httpClient must be specified")
 	}
-	return &revocation{
-		httpClient:       httpClient,
-		certChainPurpose: ocsp.PurposeCodeSigning,
-	}, nil
+
+	return NewWithOptions(context.Background(), &Options{
```

Comment:
As mentioned above, `New` should also use the caller's context.

Reply:
We will need to deprecated `New`, remove `NewTimestamp`, and add `NewWithOptions` before v1.1.0 stable release.